pkg/oci/utils.go

```diff
@@ -37,6 +37,41 @@ var (
 
 	// BundlePathKey is the annotation key to fetch the OCI configuration file path.
 	BundlePathKey = "com.github.containers.virtcontainers.pkg.oci.bundle_path"
+
+	// ContainerTypeKey is the annotation key to fetch container type.
+	ContainerTypeKey = "com.github.containers.virtcontainers.pkg.oci.container_type"
+
+	// CRIOContainerTypeKey is the annotation key to fetch CRIO container type.
+	// It is referenced in CRI-O code in two different places:
+	// - https://github.com/kubernetes-incubator/cri-o/blob/v0.3/server/sandbox_run.go#L260
+	// - https://github.com/kubernetes-incubator/cri-o/blob/v0.3/server/container_create.go#L478
+	CRIOContainerTypeKey = "crio/container_type"
+
+	// CRIOSandboxNameKey is the annotation key to fetch CRIO sandbox/pod name.
+	// It is referenced in CRI-O code here:
+	// - https://github.com/kubernetes-incubator/cri-o/blob/v0.3/server/container_create.go#L477
+	CRIOSandboxNameKey = "crio/sandbox_name"
+
+	// ContainerTypeContainer defines a regular container.
+	// Keep in sync with CRI-O, see following URL for reference:
+	// https://github.com/kubernetes-incubator/cri-o/blob/v0.3/server/container.go#L13
+	ContainerTypeContainer = "container"
+
+	// ContainerTypePod defines a sandbox/pod container.
+	// Keep in sync with CRI-O, see following URL for reference:
+	// https://github.com/kubernetes-incubator/cri-o/blob/v0.3/server/container.go#L11
+	ContainerTypePod = "sandbox"
+)
+
+// ContainerType defines a type of container.
+type ContainerType string
+
+// List different types of containers
+var (
+	RegularContainer     ContainerType = "regular_container"
```

Comment:
For the delete case, look at the code if there was no `oci.RegularContainer` definition and use `oci.PodSandbox` instead. It should just work, and if --force is not provided, it will prevent a Docker container or a CRI sandbox from being deleted/removed.